addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,216 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
     public void importCollection(String collection) throws JsonProcessingException {
         PostmanCollection postmanCollection = parse(collection);
 
-        // TODO: Extract list of HttpMessage from PostmanCollection and send requests
+        List<HttpMessage> httpMessages = new ArrayList<>();
+        extractHttpMessages(postmanCollection.getItem(), httpMessages);
+
+        requestor.run(httpMessages);
     }
 
     public PostmanCollection parse(String collectionJson) throws JsonProcessingException {
         ObjectMapper objectMapper = new ObjectMapper();
         return objectMapper.readValue(collectionJson, PostmanCollection.class);
     }
 
+    private void extractHttpMessages(List<AbstractItem> items, List<HttpMessage> httpMessages) {
+        for (AbstractItem item : items) {
+            if (item instanceof Item) {
+                HttpMessage httpMessage = extractHttpMessage((Item) item);
+                if (httpMessage != null) {
+                    httpMessages.add(httpMessage);
+                }
+            } else if (item instanceof ItemGroup) {
+                extractHttpMessages(((ItemGroup) item).getItem(), httpMessages);
+            }
+        }
+    }
+
     private static boolean isSupportedScheme(String scheme) {
         return "http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme);
     }
+
+    public static boolean isContentTypeAlreadySet(List<KeyValueData> headers) {
+        for (KeyValueData header : headers) {
+            if (header.getKey().equalsIgnoreCase("Content-Type")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static HttpMessage extractHttpMessage(Item item) {
+        Request request = item.getRequest();
+        Url url = request.getUrl();
+
+        if (request == null || url == null) {
+            return null;
+        }
+
+        HttpMessage httpMessage;
+        try {
+            String rawUrl = url.getRaw();
+            httpMessage = new HttpMessage(new URI(rawUrl, false));
+        } catch (URIException | HttpMalformedHeaderException | NullPointerException e) {
+            return null;
+        }
+
+        httpMessage.getRequestHeader().setMethod(request.getMethod());
+
+        for (KeyValueData header : request.getHeader()) {
+            if (!header.isDisabled()) {
+                httpMessage.getRequestHeader().setHeader(header.getKey(), header.getValue());
+            }
+        }
+
+        Body body = request.getBody();
+
+        if (body.isDisabled()) {
+            return httpMessage;
+        }
+
+        String mode = body.getMode();
+
+        String bodyContent = "";
+        String contentType = "";
+
+        if (mode == null || body == null) {
+            return httpMessage;
+        }
+        if (mode.equals("raw")) {
```

Comment:
Worth extracting constants, though I guess we could use an enum (if the latter better in other PR though).

Reply:
I created constants as we also needed to check if the given mode is in the allowed modes.